pageserver/src/tenant.rs

```diff
@@ -5618,12 +5618,27 @@ mod tests {
                 updated[blknum] = lsn;
             }
 
-            // Perform a cycle of flush, compact, and GC
-            tline.freeze_and_flush().await?;
-            tline.compact(&cancel, EnumSet::empty(), &ctx).await?;
-            tenant
-                .gc_iteration(Some(tline.timeline_id), 0, Duration::ZERO, &cancel, &ctx)
-                .await?;
+            // Perform two cycles of flush, compact, and GC
```

Comment:
Can you add a check that compaction actually did what you expect?

Reply:
If it doesn't crash + runs in a reasonable amount of time, it works, so I don't plan to add checks for now.

Once the vectored get interface can stop when an image layer covers all keyspace, I will add a check that `vectored_access_delta_file_cnt` is decreasing for each round of compaction/force_image_layer_creation.

Reply:
> If it doesn't crash + runs in a reasonable amount of time + does not lose any data, it works, so I don't plan to add checks for now.

It would also "work" if no image layers are added.

Reply:
added a test case on l0 compaction check https://github.com/neondatabase/neon/pull/7567/commits/6eaacbf14d8fc3e5c18239296c46d0324fec7f56